### PR TITLE
p2p module

### PR DESCRIPTION
**Pull request by bufrr**

### CODE REVIEW

**Comment by ai-chen2050** [comment]
@bufrr 
- Some minor conflicts in Cargo.toml file.  
- You could fix it & directly merge to sprint_01.
- Only must review code when merging to poc1 or master branches.
- By the way, notice squash some chore commit logs when push. etc, `update, error handle, & update`

File: crates/protos/src/lib.rs [hunk 1]
```diff
@@ -0,0 +1,17 @@
+pub mod message;
+
+pub fn add(left: usize, right: usize) -> usize {
```
Comment (ai-chen2050) [inline crates/protos/src/lib.rs:3]: This function whether is called ?

Comment (bufrr) [reply crates/protos/src/lib.rs:3]: This is auto-generated by cargo when calling `cargo new --lib`

Comment (ai-chen2050) [reply crates/protos/src/lib.rs:3]: If that's not needed, maybe consider removing it.

**Pull request merged**